pkg/client/cloudevents/grpcsource/watch.go

```diff
@@ -18,18 +19,26 @@ type workWatcher struct {
 	done    chan struct{}
 	stopped bool
 
+	ctx    context.Context
```

Comment:
since we have conext for the watcher, can we delete `done` channel? they seem duplicated.

Reply:
`done` for  explicitly stop watch